android/tangram/src/com/mapzen/tangram/MapData.java

```diff
@@ -0,0 +1,102 @@
+package com.mapzen.tangram;
+
+import java.util.List;
+
+public class MapData {
+
+    /**
+     * Construct a new MapData object for adding drawable data to the map
+     * @param name Name of the data source in the scene file for styling this object's data
+     */
+    public MapData(String name) {
+        this.name = name;
+        this.id = addDataSource(name);
+    }
+
+    /**
+     * Get the name of this data source
+     * @return The name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Remove all data from this source
+     * @return This object, for chaining
+     */
+    public MapData clear() {
+        clearSourceData(id);
+        return this;
+    }
+
+    /**
+     * Add geometry from a GeoJSON string to this data source
+     * @param data String of GeoJSON containing a Feature or FeatureCollection
+     * @return This object, for chaining
+     */
+    public MapData addGeoJSON(String data) {
+        addSourceGeoJSON(id, data);
+        return this;
+    }
+
+    /**
+     * Add a point geometry to this data source
+     * @param point LngLat with the coordinates of the point
+     * @return This object, for chaining
+     */
+    public MapData addPoint(LngLat point) {
```

Comment:
We want to be pretty generic about the data that can be associated with a piece of geometry, but now that I think about it using something as simple as an integer identifier wouldn't necessarily constrain us in that regard. Instead of requiring that the metadata "live" inside tangram, we let the users store the metadata in whatever way is most convenient and when they select a feature, we simply provide an identifier that enables them to query metadata themselves. This would probably be far more convenient for developers, but it diverges a bit from the way that we've been thinking about the results of a feature query.